doc/classes/Object.xml

```diff
@@ -1,33 +1,38 @@
 	<brief_description>
-		Base class for all non-built-in types.
+		Base class for all objects in the engine.
 	</brief_description>
 	<description>
-		Every class which is not a built-in type inherits from this class.
-		You can construct Objects from scripting languages, using [code]Object.new()[/code] in GDScript, or [code]new Object[/code] in C#.
-		Objects do not manage memory. If a class inherits from Object, you will have to delete instances of it manually. To do so, call the [method free] method from your script or delete the instance from C++.
-		Some classes that extend Object add memory management. This is the case of [RefCounted], which counts references and deletes itself automatically when no longer referenced. [Node], another fundamental type, deletes all its children when freed from memory.
-		Objects export properties, which are mainly useful for storage and editing, but not really so much in programming. Properties are exported in [method _get_property_list] and handled in [method _get] and [method _set]. However, scripting languages and C++ have simpler means to export them.
-		Property membership can be tested directly in GDScript using [code]in[/code]:
+		An advanced [Variant] type. All classes in the engine inherit from Object. Each class may define new properties, methods or signals, which are available to all inheriting classes. For example, a [Sprite2D] instance is able to call [method Node.add_child] because it inherits from [Node].
+		You can create new instances, using [code]Object.new()[/code] in GDScript, or [code]new Object[/code] in C#.
+		To delete an Object instance, call [method free]. This prevents memory leaks, and is necessary for most classes inheriting Object, because they do not manage memory on their own. [RefCounted] is the biggest exception to this. [Node], another essential object, also deletes its children when freed.
+
+		Objects can have a [Script] attached to them. Once the [Script] is instantiated, it effectively acts as an extension to the base class, allowing it to define and inherit new properties, methods and signals.
+		Inside a [Script], virtual methods can be overriden to change behavior. In particular, [method _get_property_list] may be overriden to customize properties in several ways. This allows them to be available to the editor, display as lists of options, sub-divide into groups, save on disk, etc. Scripting languages offer easier ways to customize properties, such as with the [annotation @GDScript.@export] annotation.
+		Because Godot is very dynamic, there can be occasions where, for example, a property may not exist. To prevent run-time errors, see methods such as [method set], [method get], [method call], [method has_method], [method has_signal] etc. Note that these methods are much slower than a direct reference to the members. In GDScript, you can also check if a given property, method, or signal name exists in an object with the [code]in[/code] operator:
```

Comment:
It might be more helpful to say what makes Godot/Object dynamic, rather than that it simply is dynamic. Something like Object properties and methods being able to change at runtime to fit app/game needs, and so it is useful to have ways to check and set those properties, using the the methods you're already listing!

Reply:
Definitely better, though for newbies (like meee) explicitly saying that an Object's properties can change at runtime is helpful in that it just settles the issue for the reader. That way it doesn't have to be inferred from the fact that 1) the script defines the properties and the script can change, and 2) the _set and _get methods actually exist!

Reply:
I am struggling to understand this is a positive comment or not, could you elaborate?

Reply:
I think what I'm saying is that I'd personally find useful to see this change:

`Godot is very dynamic. An object's script can be changed at run-time, and all objects can be passed into any method with Object parameters.`

to

`Godot is very dynamic. An object's script, and therefore its properties and methods, can be changed at run-time.`

This says the properties and methods change at runtime explicitly, which I think is helpful to assert early and clearly. I think the part about objects isn't necessary, since it's not specific to Godot, just programming languages.

Hopefully that clarified it!

Reply:
It did very much. I modified the description once more with something similar to your sentence.

Reply:
Looks wonderful! <3